### PR TITLE
Fixes some bugs for drawing classifier

### LINKED ISSUE #1918: Confusing console output on drawing_classifier.evaluate(...)
```python
In [17]: len(test_data)
Out[17]: 143

In [18]: _ = model.evaluate(test_data)
Predicting 144/143

In [19]: len(longer_test_data)
Out[19]: 429

In [20]: _ = model.evaluate(longer_test_data)
Predicting 257/429
```
The output from the first call to `evaluate(...)` is really the most confusing. How is it predicting one more examples than was passed in? This seems like an off by one bug.

The output from the second call to `evaluate(...)` makes it seem like it never finishes. The last/only output makes it seem like it's only about half done. The final output should probably be `Predicting X/X` where `X` is the number of examples passed in.

Both calls to evaluate finish quickly. I would recommend only outputting progress info if it's been at least 30 seconds.

### LINKED ISSUE #1919: log_loss missing from drawing classifier evaluation metrics
The dictionary returned by `DrawingClassifier.evaluate(...)` does not contain an entry for `log_loss`. Most (all?) of the other classifiers return a log loss in their evaluation metrics. For consistency, it should be added for the drawing classifier.

### LINKED ISSUE #1931: Log shows one less iteration than specified max_iterations
```
In [3]: model = tc.drawing_classifier.create(data,"label","bitmap", warm_start="auto", validation_set=None, max_iterations=10)
+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+
| 1                      | 2.931                  | 0.231                  | 17.959                 |
+------------------------+------------------------+------------------------+------------------------+
| 2                      | 2.631                  | 0.448                  | 40.837                 |
+------------------------+------------------------+------------------------+------------------------+
| 3                      | 2.417                  | 0.618                  | 63.040                 |
+------------------------+------------------------+------------------------+------------------------+
| 4                      | 2.388                  | 0.674                  | 85.326                 |
+------------------------+------------------------+------------------------+------------------------+
| 5                      | 2.354                  | 0.685                  | 106.944                |
+------------------------+------------------------+------------------------+------------------------+
| 6                      | 2.397                  | 0.690                  | 128.826                |
+------------------------+------------------------+------------------------+------------------------+
| 7                      | 2.449                  | 0.701                  | 150.282                |
+------------------------+------------------------+------------------------+------------------------+
| 8                      | 2.364                  | 0.725                  | 172.006                |
+------------------------+------------------------+------------------------+------------------------+
| 9                      | 2.351                  | 0.727                  | 193.263                |
+------------------------+------------------------+------------------------+------------------------+
```

### LINKED ISSUE #1941: No validation accuracy should be represented by None not nan
If you train a drawing classifier without a validation set, the validation accuracy gets saved as a `nan`. I think `None` makes a lot more sense.

### LINKED ISSUE #1943: Drawing Classifier should not save batch_size
The drawing classifier saves the `batch_size` values given in the `create(...)` call to its state. It's not clear how/if this is used since `evaluate(...)`, `predict(...)` and `predict_topk(...)` all take a `batch_size` parameter.

In generally we don't want to store batch size in model state. Batch size isn't really model dependent but dependent on the machine it is running on.

### PR DESCRIPTION
**Pull request by shreyajain17**
fixes #1941
fixes #1943
fixes #1919
fixes #1931

### CODE REVIEW

**Comment by shreyajain17** [comment]
The first epoch was being skipped by mistake but I still feel like the time elapsed for the first iteration is suspicious.
```
model = tc.drawing_classifier.create(data,"label","bitmap", warm_start="auto", validation_set=None, max_iterations=4)
+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+
| 1                      | 2.968                  | 0.056                  | 0.544                  |
+------------------------+------------------------+------------------------+------------------------+
| 2                      | 2.941                  | 0.231                  | 24.502                 |
+------------------------+------------------------+------------------------+------------------------+
| 3                      | 2.721                  | 0.449                  | 46.366                 |
+------------------------+------------------------+------------------------+------------------------+
| 4                      | 2.468                  | 0.622                  | 68.647                 |
+------------------------+------------------------+------------------------+------------------------+
```
Now when I type model.validation_accuracy (changed it to None) in ipython I get nothing. Is it alright?

Batch size is not stored as a state parameter.

Log_loss is being given in metrics. Unfortunately I had forgotten we may or may not keep it ultimately.

**Comment by TobyRoseman** [comment]
> The first epoch was being skipped by mistake but I still feel like the time elapsed for the first iteration is suspicious.
> 
> ```
> model = tc.drawing_classifier.create(data,"label","bitmap", warm_start="auto", validation_set=None, max_iterations=4)
> +------------------------+------------------------+------------------------+------------------------+
> | Iteration              | Training Loss          | Training Accuracy      | Elapsed Time (seconds) |
> +------------------------+------------------------+------------------------+------------------------+
> | 1                      | 2.968                  | 0.056                  | 0.544                  |
> +------------------------+------------------------+------------------------+------------------------+
> | 2                      | 2.941                  | 0.231                  | 24.502                 |
> +------------------------+------------------------+------------------------+------------------------+
> | 3                      | 2.721                  | 0.449                  | 46.366                 |
> +------------------------+------------------------+------------------------+------------------------+
> | 4                      | 2.468                  | 0.622                  | 68.647                 |
> +------------------------+------------------------+------------------------+------------------------+
> ```

I agree this is very suspicious.

> Now when I type model.validation_accuracy (changed it to None) in ipython I get nothing. Is it alright?

That is expected. That's what happens when you evaluate `None` in ipython.

> Log_loss is being given in metrics. Unfortunately I had forgotten we may or may not keep it ultimately.

Why might we not keep it?

**Comment by shreyajain17** [comment]
> Why might we not keep it?

Aren't we discussing on another issue to maybe remove it completely.

**Comment by nickjong** [comment]
The `evaluate` and `predict` methods shouldn't take a batch size at all. Let's be consistent with OD and AC for now
